src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXLegalityCheck.cpp

```diff
@@ -682,7 +682,7 @@ bool isSuitableForZDNN<ONNXMaxPoolSingleOutOp>(ONNXMaxPoolSingleOutOp op) {
          "Failed to scan ONNXMaxPoolSingleOutOp parameters successfully");
 
   // dilations not supported. Only default one is accepted.
-  if (shapeHelper.dilations[0] != 1 || shapeHelper.dilations[1] != 1)
+  if (llvm::any_of(shapeHelper.dilations, [](int64_t val) { return val != 1; }))
```

Comment:
zDNN pooling ops support 4D tensors only, so it always has dilations[1]. You just need to call `checkLegalityPoolOpsCommon` (which checks tensor rank, etc. ) before this check to fix the issue.

Reply:
Done. Thanks!